api4/job.go

```diff
@@ -167,7 +167,12 @@ func getJobs(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	w.Write([]byte(model.JobsToJson(jobs)))
+	js, jsonErr := json.Marshal(jobs)
+	if jsonErr != nil {
+		c.Err = model.NewAppError("getJobs", "api.marshal_error", nil, jsonErr.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Write(js)
```

Comment:
What was the reasoning behind using this approach vs. `json.NewDecoder(w)`?

Reply:
No strong reason, just wanted to separate the error type and make sure that if we fail with that code is due to marshaling and not some other error writing the response. We can change it later on to make it consistent over all endpoints.